fulltext/backends/__csv.py

```diff
@@ -20,9 +34,14 @@ def _get_file(f, **kwargs):
     elif mimetype == 'text/psv':
         options['delimiter'] = '|'
 
-    text, f = StringIO(), codecs.getreader('utf8')(f, errors='ignore')
+    if PY3:
+        gen = readlines(f, encoding, errors)
+        reader = csv.reader(gen, **options)
+    else:
+        reader = unicode_reader(f, encoding, errors, **options)
```

Comment:
Can you define unicode_reader() for PY3 as well, so both paths are equivalent?

```python
if PY3:
    def unicode_reader(...):
        def _readlines(...):
            ...
        ...

else
    def unicode_reader(...):
        ...

def _get_file(...):
    reader = unicode_reader(f, encoding, errors, **options)```

Reply:
sure, done